### PR TITLE
Add samples tests for Fun0ExpectationSamples in infix syntax

### PR DESCRIPTION
**Pull request by Kushiro-C**
Hello, I think I finished the issue [#1044](https://github.com/robstoll/atrium/issues/1044#issue-1045235262l)
 and would like to have a review on if there is anything wrong, or anything that could be improved.
______________________________________
I confirm that I have read the [Contributor Agreements v1.0](https://github.com/robstoll/atrium/blob/main/.github/Contributor%20Agreements%20v1.0.txt), agree to be bound on them and confirm that my contribution is compliant.

### CODE REVIEW

File: apis/infix-en_GB/atrium-api-infix-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/samples/Fun0ExpectationSamples.kt [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    @Test
+    fun toThrowFeature() {
+        expect { throw IllegalStateException("abc") }
+            .toThrow<IllegalStateException>() messageToContain "abc"
+        //      | subject is now of type IllegalStateException
+
+        fails {
+            expect { throw IllegalStateException("abc") }
+                .toThrow<IndexOutOfBoundsException>() messageToContain "abc"
+            // not shown in reporting as `toThrow<IndexOutOfBoundsException>()` already fails
```
Comment (robstoll) [inline apis/infix-en_GB/atrium-api-infix-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/samples/Fun0ExpectationSamples.kt:18]:
```suggestion
            expect { throw IllegalStateException("abc") }
                .toThrow<IndexOutOfBoundsException>() messageToContain "abc"
            //                                        | not shown in reporting as `toThrow<IndexOutOfBoundsException>()` already fails
```

File: apis/infix-en_GB/atrium-api-infix-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/samples/Fun0ExpectationSamples.kt [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        }
+    }
+
+    @Test
+    fun toThrow() {
+        expect { throw IllegalStateException("abc") }
+            .toThrow<IllegalStateException> { // subject inside this block is of type IllegalStateException
+                it messageToContain "abc"
```
Comment (robstoll) [inline apis/infix-en_GB/atrium-api-infix-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/samples/Fun0ExpectationSamples.kt:26]: IMO reads a bit nicer but also fine to keep it this way
```suggestion
                its messageToContain "abc"
```

File: apis/infix-en_GB/atrium-api-infix-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/samples/Fun0ExpectationSamples.kt [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            } // subject keeps type IllegalStateException also after the block
+
+        fails { // because wrong type expected (IndexOutOfBoundsException instead of IllegalStateException), but since we use a block...
+            expect { throw IllegalStateException("abc") }
+                .toThrow<IndexOutOfBoundsException> {
+                    it messageToContain "abc" // ... reporting mentions that subject's message was expected `to contain: "abc"`
```
Comment (robstoll) [inline apis/infix-en_GB/atrium-api-infix-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/samples/Fun0ExpectationSamples.kt:32]:
```suggestion
                    its messageToContain "abc" // ... reporting mentions that subject's message was expected `to contain: "abc"`
```

**Review by robstoll — APPROVED** [review]
@Kushiro-C looks good, thanks. Three small suggestions but up to you if you want to improve it or not. Let me know what you think

**Comment by Kushiro-C** [comment]
Hello @robstoll , thank you for your review.
I agree with your suggestions, and have already pushed a commit adding them.
Please let me know if there is anything else to modify.

**Review by robstoll — APPROVED** [review]

**Comment by robstoll** [comment]
@Kushiro-C thanks for your first contribution to Atrium 🙂👍
How about a second? 😉

**Comment by Kushiro-C** [comment]
Thank you ☺️

> How about a second? 😉


I'll think about it when my schedule is more open 🙂

**Pull request merged**